packages/rest/src/lib/RequestManager.ts

```diff
@@ -404,14 +406,17 @@ export class RequestManager extends EventEmitter {
 			}
 		}
 
-		const fetchOptions = {
-			agent: this.agent,
-			body: finalBody,
+		const fetchOptions: RequestInit = {
+			body: finalBody!,
```

Comment:
This assertion is under false pretense at first glance, you should leave a comment here explaining its actual purpose

Reply:
fixed in c8fdd310579c6473fd5f0423c04dd636ec2d5fce